server/app/controllers/ErrorHandler.java

```diff
@@ -55,11 +60,12 @@ public CompletionStage<Result> onServerError(RequestHeader request, Throwable ex
    * <p>The framework provides wrapped exceptions to the methods in this class so we have to dig out
    * our application exception. Anecdotally it's 2 levels down.
    */
-  static Optional<Throwable> findThrowableByType(Throwable exception, Class<?> search) {
+  static Optional<Throwable> findThrowableByTypes(
+      Throwable exception, ImmutableSet<Class<?>> search) {
     Optional<Throwable> root = Optional.of(exception);
     // Search a couple causes deep for the desired type.
     for (int i = 0; i < 5 && root.isPresent(); ++i) {
-      if (search.isInstance(root.get())) {
+      if (search.contains(root.get().getClass())) {
```

Comment:
Not-blocking: Previously, we'd match subclasses of the provided "search" parameter, whereas now it has to be an exact match.

I'd suggest documenting that and marking the two specific error types that we currently handle as "final" to prevent us trying to take any implicit dependency on that behavior.

Reply:
good call! done